infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationMetadataManager.java

```diff
@@ -99,71 +104,111 @@ public LogReplicationMetadataManager(CorfuRuntime rt, long topologyConfigId, Str
     }
 
     public CorfuStoreMetadata.Timestamp getTimestamp() {
-        return corfuStore.getTimestamp();
+        Token token = runtime.getSequencerView().query().getToken();
+        return CorfuStoreMetadata.Timestamp.newBuilder()
+                .setEpoch(token.getEpoch())
+                .setSequence(token.getSequence())
+                .build();
```

Comment:
Is this function still needed?

Reply:
So this is only needed for the snapshot sync start marker timestamp. I think we might be able to get this timestamp from somewhere else (for instance from the tx.commit of the first shadow stream commit, but got lost in the code with many other changes).. let me know your thoughts..

Reply:
Yes, from tx.commit makes more sense. As we don't want users to build their TS.

Reply:
Ok, made this change.. this is the only one I'm a bit concerned we don't see the implications of this change. If you can give it some extra thoughts, I'm doing the same.